rows.go

```diff
@@ -64,12 +65,24 @@ func (rows *mysqlRows) Columns() []string {
 	return columns
 }
 
+func (rows *mysqlRows) setFinish(f func()) {
```

Comment:
This is implementation of `setfinish` interface that defined in connection_go18.go.
Of course, I can rewrite it setting directly, but `QueryContext` will be a little complex.

``` go
// in QueryContext
switch r := rows.(type) {
    case mysqlRows:
        r.finish = mc.finish
    case binaryRows:
        r.finish = mc.finish
    case textRows:
        r.finish = mc.finish
    default:
        mc.finish()
}
```

Now implementation is here.

``` go
// in QueryContext
if set, ok := rows.(setfinish); ok {
	set.setFinish(mc.finish)
} else {
	mc.finish()
}
```